src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -787,7 +805,7 @@ private SharedPreferences getSharedPreferences() {
         private final String mToken;
         private final Lock mStartLock;
 
-        private final List<Pair<String,JSONObject>> mEditorChanges;
+        private final Map<String, Pair<String,JSONObject>> mEditorChanges;
```

Comment:
Yes. We always try to use the abstract interfaces rather than their concrete types. We don't really care if the Map we're given is based in a HashTable or a Tree or even something more exotic- all users need to know is that it's a map.

We try to do this wherever we can- only refer to types by their simplest usable superclass - but it's particularly idiomatic for collections.